lib/ZnapZend.pm

```diff
@@ -145,6 +145,216 @@ my $killThemAll = sub {
+                            last;
+                        }
+                        if ($idx < 1 || $ancestor eq $dataSet) {
+                            # Chopping did not go well
+                            last;
+                        }
+                    }
+
+                    if (defined($nearestLE)) {
+                        # Got something, is it "on" or "off", is recursion involved?
+                        if (defined($nearestLR)) {
+                            # Both properties are locally defined in some ancestor(s)
+                            if ($nearestLE eq "off" && $nearestLR eq "on" && $nearestLEds eq $nearestLRds) {
+                                # An ancestor defines both enabled==off and recursive==on
+                                # explicitly, and is the nearest one to define such things.
+                                $isDisabled = 1;
+                            }
+                            # else enabled=on, or recursive=off, or not set in same dataset
+                            elsif ($nearestLE eq "on" && $nearestLR eq "on" && $nearestLEds eq $nearestLRds) {
+                                # logical inheritance of "enabled=on"
+                                $isDisabled = -1;
+                            }
+                            # else enabled=on, or recursive=off, or not set in same dataset
+                            elsif ($nearestLE eq "on" && $nearestLR eq "off" && $nearestLEds eq $nearestLRds) {
+                                # logical inheritance of "enabled=..." from parent of that
+                                # ancestor because its own "enabled=on" is not recursive
+                                my $ancestor2 = $nearestLEds;
+                                while ($ancestor2) {
+                                    $ancestor2 =~ s/\/[^\/]+$// ;
+                                    if (grep {$_ eq $ancestor2} @dataSetsExplicitlyDisabled) {
+                                        # We handled that ancestor and found it disabled, in an earlier loop
+                                        $isDisabled = 1;
+                                        last;
+                                    }
+                                    if (index($ancestor2, '/') < 0) {
+                                        last;
+                                    }
+                                }
+                            }
+                            elsif ($nearestLE eq "on" && defined($propIR) && $propIR eq "on") {
+                                # zfs-inheritance of "enabled=on"
+                                $isDisabled = -1;
+                            }
+                        }   # else only "enabled=whatever" is defined in some ancestor, but
+                            # not "recursive" anywhere => this one remains enabled, probably
+                    }   # else no ancestor defines a local "enabled" setting
```

Comment:
you are making use of && a lot here instead of the looser `and` intentional ?

Reply:
Rather, forgot the difference - hopping too many languages lately :) Do you suggest I change these?

Reply:
well I am normally weary of using && for its high binding power

Reply:
PR update posted